lib/OptimizationOptimJL/src/OptimizationOptimJL.jl

```diff
@@ -128,7 +136,8 @@ function SciMLBase.__solve(cache::OptimizationCache{
     local x, cur, state
 
     cur, state = iterate(cache.data)
-
+    (requiresgradient(Optim.AbstractOptimizer))
```

Comment:
This should be done in the `solve` defined in SciMLBase, I'll point you to it later once this PR is done, you can remove this error from here